### PR TITLE
Dry run

### PR DESCRIPTION
**Pull request by dcrystalj**

### CODE REVIEW

File: actions/connect.go [hunk 1]
```diff
@@ -84,31 +86,38 @@ func (c *Connector) communicate(ctx context.Context, stream actionStreamer) erro
 			case msg.Sequence == api.Sequenece_EXECUTE:
 				plugin, ok := c.Plugins[msg.Action]
 				if !ok {
-					stream.Send(&api.AgentReply{
+					err = stream.Send(&api.AgentReply{
 						JobId:    msg.JobId,
 						Sequence: api.Sequenece_EXECUTE,
```
Comment (fiksn) [inline actions/connect.go:91]: Sequenece_EXECUTE, typo in `sequence`

File: actions/connect_test.go [hunk 2]
```diff
@@ -92,15 +99,70 @@ func TestCommunicate(t *testing.T) {
...
+		ctx, cancel := context.WithCancel(context.Background())
+		bs := blockingStream{
+			msg: &api.Action{
+				JobId:    5,
+				Action:   "test",
+				Sequence: api.Sequenece_EXECUTE,
+			},
+			sent: []*api.AgentReply{},
+		}
+		c := Connector{
```
Comment (fiksn) [inline actions/connect_test.go:147]: For consistency reasons `c` should be `cc` here as well?

**Review by fiksn — APPROVED** [review]

File: actions/connect_test.go (hunk 2 above)
Comment (dcrystalj) [reply actions/connect_test.go:147]: this one is "local" thats why I changed it

File: actions/connect.go (hunk 1 above)
Comment (dcrystalj) [reply actions/connect.go:91]: nice

File: actions/connect_test.go [hunk 3]
```diff
@@ -14,9 +15,14 @@ import (
 	"github.com/urfave/cli"
 )
 
-type testPlugin struct{}
+type testPlugin struct {
+	shouldFail bool
+}
 
-func (tp *testPlugin) Execute(jobID int32, data []byte, msg func(entities.PluginMessage) error, isDryRun bool) error {
+func (tp *testPlugin) Execute(jobID int32, data []byte, msg func(entities.PluginMessage) error) error {
```
[the remark this answers is not in the document]
Comment (dcrystalj) [reply actions/connect_test.go]: will change in other pr

**Pull request merged**